dask_expr/_collection.py

```diff
@@ -400,6 +400,9 @@ def __getitem__(self, other):
             other = other.copy()
         return new_collection(self.expr.__getitem__(other))
 
+    def __dask_tokenize__(self):
+        return type(self).__name__, self._expr._name
```

Comment:
The collection is also being tokenized on compute. Don't know exactly why but it's been using the repr so far which helped me find https://github.com/dask-contrib/dask-expr/pull/941